src/MenuButton/index.js

```diff
@@ -35,9 +31,11 @@ const MenuButton = ({
   alignment = "start",
   offset = 2,
   children,
+  footerItem,
 }) => {
   const [isOpen, setIsOpen] = useState(false);
   const menuItems = React.Children.toArray(children);
+  const allItems = footerItem ? menuItems.concat(footerItem) : menuItems;
```

Comment:
For the subcomponent interface it would be more like...

```jsx
const allItems = React.Children.toArray(children);
const menuItems = allItems.filter(isMenuItem); // see implementation of filter in other downshift components
const footerItem = allItems.find(isFooterItem); // take the first MenuButton.Footer in children and use that
```

I don't feel strongly about this though. Using `footerItem` might be more consistent with the model we use in other components with a pinned footer, e.g. `Dialog`